src/classes/shex-form-model.ts

```diff
@@ -0,0 +1,266 @@
+import N3, { DataFactory, Store, Writer } from 'n3'
+import { NS_RDF, NS_DC, NS_UI, NS_LAYOUT, IRI_XsdString } from '../constants'
+import { ListObject } from './list-object'
+import { Meta } from '../interfaces'
+/**
+ * Convert ShEx to Tim Form Model
+ */
+export class ShexFormModel {
+  termFactory: any
+  meta: Meta
+  schema: any
+  graph: any
+  iriRdftype: string
+  iriDctitle: string
+  iriUitype: string
+
+  constructor(schema: any) {
+    this.termFactory = DataFactory
+    this.graph = new Store()
+    this.meta = { prefixes: schema._prefixes, base: window.location.href }
+    this.iriRdftype = `${NS_RDF}type`
+    this.iriDctitle = `${NS_DC}title`
+    this.iriUitype = this.iriRdftype
+    this.schema = schema
+  }
+  /**
+   * Convert SheEx to Form Model
+   */
+  convert() {
+    const {
+      schema,
+      graph,
+      termFactory: { namedNode, blankNode, literal }
+    } = this
+    const IRI_this = '#'
+    /**
+     * Get root for term into ShexEx
+     */
+    const rootFormTerm = namedNode(`${IRI_this}formRoot`)
+    /**
+     * Find ShEx expression
+     */
+    const start =
+      'start' in this.schema ? this.derefShapeExpression(schema.start) : schema.shapes[0]
+
+    /**
+     * Run into shape to create turtle object
+     */
+    this.walkShape(start, rootFormTerm, this.localName(start.id), namedNode, literal, blankNode)
+
+    const writer = new Writer({
+      prefixes: { '': IRI_this, ui: NS_UI, dc: NS_DC },
+      listHeads: graph.sequesterLists()
+    })
+    writer.addQuads(graph.getQuads())
+    let formModel
+    writer.end((error, result) => (formModel = result))
+
+    return formModel
+  }
+  /**
+   * Find label expression into ShEx
+   * @param shexpr
+   */
+  findTitle(shexpr: any) {
+    return (shexpr.annotations || []).find((a: any) => a.predicate === this.iriDctitle)
+  }
+
+  /**
+   * Define shape type into formModel
+   * @param iri
+   */
+  localName(iri: string) {
+    const { meta } = this
+
+    if (iri.startsWith('_:')) {
```

Comment:
Please add a comment explaining why we're doing this (I assume it's catching blank nodes?)